File: src/main/java/com/edureka/customer/service/CustomerService.java
package com.edureka.customer.service;

import java.util.List;

import com.edureka.customer.dto.CustomerDto;

public interface CustomerService {

	 List<CustomerDto> listCustomers();
	
	 CustomerDto addCustomer(CustomerDto customerDto);
	
	 CustomerDto customerDetails(String customerId);

}

File: src/main/resources/static/js/app/customerService.js
'use strict'

angular.module('customer.services', []).factory('customerService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getCustomerById = function(customerId) {
				var url = CONSTANTS.getCustomerById + customerId;
				return $http.get(url,'application/json');
			}
			service.getAllCustomers = function() {
				return $http.get(CONSTANTS.getAllCustomers,'application/json');
			}
			service.addCustomer = function(customerDto) {
				return $http.post(CONSTANTS.addCustomer, customerDto);
			}
			return service;
		} ]);